### PR TITLE
Selective vehicle claxon

### PR DESCRIPTION
**Pull request by spacek531**
To (hopefully) solve https://github.com/OpenRCT2/OpenRCT2/issues/7657

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -10050,5 +10050,12 @@ void vehicle_update_crossings(const rct_vehicle * vehicle)
 void vehicle_claxon(const rct_vehicle * vehicle)
 {
     rct_ride_entry* rideEntry = get_ride_entry(vehicle->ride_subtype);
-    audio_play_sound_at_location((rideEntry->vehicles[0].sound_range == 3)? SOUND_TRAIN_WHISTLE : SOUND_TRAM, vehicle->x, vehicle->y, vehicle->z);
+    Switch (rideEntry->vehicles[vehicle->vehicle_type].sound_range) {
```
Comment (AaronVanGeffen) [inline src/openrct2/ride/Vehicle.cpp]: Should be all-lowercase, i.e. `switch`.

Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.cpp]:
```
switch (rideEntry->vehicles[vehicle->vehicle_type].sound_range) 
{
```

**Comment by Gymnasiast** [comment]
Looks good to me, just a formatting thingy to do.

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -10050,5 +10050,12 @@ void vehicle_update_crossings(const rct_vehicle * vehicle)
...
+    case 3:
```
Comment (AaronVanGeffen) [inline src/openrct2/ride/Vehicle.cpp]: We should probably introduce some constants for these…

**Comment by spacek531** [comment]
Probably why I shouldn't try to program without my glasses... or at 03:00

File: src/openrct2/ride/Vehicle.cpp [hunk 3]
```diff
@@ -10052,10 +10052,10 @@ void vehicle_claxon(const rct_vehicle * vehicle)
     rct_ride_entry* rideEntry = get_ride_entry(vehicle->ride_subtype);
     switch (rideEntry->vehicles[vehicle->vehicle_type].sound_range)
     {
-    case 3:
+    case SOUND_TRAIN_WHISTLE:
```
Comment (AaronVanGeffen) [inline src/openrct2/ride/Vehicle.cpp]: This isn't going to work — these constants represent the sound files themselves. Their values are much higher than 3 and 4.

Comment (spacek531) [reply src/openrct2/ride/Vehicle.cpp]: I looked in vehicle.h to try and find the numbers when I first started this PR. I didn't find them there so I thought they didn't exist.

Comment (AaronVanGeffen) [reply src/openrct2/ride/Vehicle.cpp]: Indeed, they probably don't exist yet.

Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp]: You need to create new constants.

```
enum sound_range
{
    SOUND_RANGE_TRAIN = 3,
    SOUND_RANGE_TRAM = 4,
}
```

**Comment by spacek531** [comment]
Is no name preferable then? I copied what gymnasiast wrote in his comment and added the enums I found in X123M3-256's ride generator, but the other enums in vehicle.h are not named.

**Comment by Gymnasiast** [comment]
Yeah, I was mistaken about lowercase/uppercase. Broxzier is right.

File: src/openrct2/ride/Vehicle.h [hunk 4]
```diff
@@ -402,8 +402,7 @@ enum {
     VEHICLE_SOUND2_FLAGS_LIFT_HILL = 1 << 1 // When on a lift hill generate sound
 };
 
-enum friction_sound
-{
+enum {
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.h]: Why did you change the brace placement? According to our code style, braces should be on their own lines.

Comment (spacek531) [reply src/openrct2/ride/Vehicle.h]: it was based on other nameless enums in vehicle.h
